src/com/palmergames/bukkit/util/PluginSupportUtil.java

```diff
@@ -0,0 +1,217 @@
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+/**
+ * Allows testing support for known unsupported classes
+ * @author GNosii
+ */
+@ApiStatus.Internal
+public class PluginSupportUtil {
+
+	private static final Map<String, Support> TESTS = getSupportData();
+
+	/**
+	 * Run tests defined in {@link #getSupportData()}
+	 * @return map with classes detected and their support data
+	 */
+	public static HashMap<String, Support> test() {
+		final HashMap<String, Support> map = new HashMap<>();
+		
+		TESTS.forEach((test, support) -> {
+			if (test.startsWith("platform:")) {
+				if (Bukkit.getName().contains(cleanTest(test))) {
+					map.put(cleanTest(test), support);
+				}
+			} else if (test.startsWith("plugin:")) {
+				if (Bukkit.getPluginManager().getPlugin(cleanTest(test)) != null) {
+					map.put(cleanTest(test), support);
+				}
+			} else if (test.startsWith("economy:")){
+				if (TownyEconomyHandler.getVersion().contains(cleanTest(test))) {
+					map.put(cleanTest(test), support);
+				}
+			} else {
+				// Always try to avoid these tests!
+				try {
+					Class.forName(test);
+					map.put(test, support);
+				} catch (final ClassNotFoundException ignored) {}
+			}
+		});
+		
+		return map;
+	}
+
+	/**
+	 * Defines which classes should be tested for, and describe their support level and description.
+	 * @return hard-coded map with the qualified name for the class and the support data.
+	 */
+	private static Map<String, Support> getSupportData() {
+		final Map<String, Support> map = new WeakHashMap<>();
+		
+		/* 
+		 * To add an class to this list, copy and paste its qualified name:
+		 * map.put("com.palmergames.bukkit.towny.Towny", new Support(SupportType.EXTENSION, "Optional description."));
+		 * 
+		 * If the plugin loads before Towny does, and it's available on Bukkit's PluginManager you can use this syntax:
+		 * map.put("plugin:Towny", new Support(SupportType.EXTENSION)
+		 */
+		
+		map.put("plugin:MDCQuestioner", new Support(SupportType.UNNECESSARY, "Towny no longer requires Questioner for questions and you can safely remove this plugin."));
+		map.put("plugin:TownyNameUpdater", new Support(SupportType.UNNECESSARY, "Towny no longer depends on TownyNameUpdater for username changes and you can safely remove this plugin."));
+		 
+		map.put("plugin:floodgate", new Support(SupportType.UNSUPPORTED_PLUGIN, "Floodgate is known to cause issues regarding their username format."));
+		
+		map.put("economy:Essentials Economy", new Support(SupportType.UNSUPPORTED_ECONOMY, "Essentials Economy is known to reset town/nation balances on rare occasions. Be careful if you're using Essentials Economy."));
+		map.put("economy:EssentialsX Economy", new Support(SupportType.UNSUPPORTED_ECONOMY, "EssentialsX Economy is known to reset town/nation balances on rare occasions. Be careful if you're using EssentialsX Economy."));
```

Comment:
Probably also needs indentation.

Reply:
These indents should probably be added by the code that loops over these elements and prints them, rather than being hard coded.